heap.c

```diff
@@ -76,7 +76,7 @@ static	void	*heap_extend(const int incr)
     static char *bounds_p = block_o_bytes + sizeof(block_o_bytes);
     static char *block_p = block_o_bytes;
     
-    if (block_p + incr >= bounds_p) {
```

Comment:
I think this is right.  If we use up to bounds_p then there was enough space.

Reply:
No, here is an extreme example suppose INTERNAL_MEMORY_SPACE and incr both are 4KB. Then the first call heap_extend will fail due to block_p + 4KB == bounds_p even there is still 4KB free memory in the pool.

Reply:
But if block_p + 4096 is == to bounds_p then it 's going to return `SBRK_ERROR` right?  Am I missing something?  I think it should return `SBRK_ERROR` if it is > bounds_p only.

Reply:
Yes, my change do what you describe:
The old condition(```block_p + incr >= bounds_p```) return ```SBRK_ERROR```, the new condtition(```block_p + incr > bounds_p```) return ```block_p```.

Reply:
Oh for goodness sake.  Sorry.  I see it now.